pandas/core/indexes/datetimelike.py

```diff
@@ -703,6 +703,12 @@ def astype(self, dtype, copy=True):
             raise TypeError(msg.format(name=type(self).__name__, dtype=dtype))
         return super(DatetimeIndexOpsMixin, self).astype(dtype, copy=copy)
 
+    @Appender(DatetimeLikeArrayMixin._time_shift.__doc__)
+    def _time_shift(self, periods, freq=None):
```

Comment:
how is this related to the rest of the PR?

Reply:
This PR changes DatetimeArrayMixin._time_shift such that it no longer pins `self.name` for Index subclasses.